### ISSUE TITLE
[Python] Unable to read from gzip stream

### ISSUE THREAD

**Issue by viernullvier** [rickardp/splitstream#2]
The splitstream module somehow doesn't work with gzip file streams. I've been unable to trace the reason for this issue since no exception is raised, it just doesn't work at all.

Workaround: Wrapping the entire file into a StringIO stream - BufferedReader doesn't work either, it seems to pass the unprocessed gzip data to splitstream.

Tested with Python 2.7.10 on OS X

``` python
from splitstream import splitfile
from cStringIO import StringIO
from io import BufferedReader
import gzip

with gzip.open("file.gz", 'rb') as f:
    for obj in splitfile(f, "json"):
        print obj  # will never get called
    f.seek(0)
    for obj in splitfile(BufferedReader(f), "json"):
        print obj  # returns garbage
    f.seek(0)
    for obj in splitfile(StringIO(f.read()), "json"):
        print obj  # works as intended
```

**Comment by rickardp** [comment]
This was actually a design choice : if there is a fileno, it is assumed that it could be used directly instead of calling into Python for each chunk to read.

However, it seems that gzip file objects also contain a fileno, which is not how I understand the [docs](https://docs.python.org/2/library/stdtypes.html?highlight=fileno#file.fileno). I could add an option to disable this optimization, but I would prefer a canonical way of determining if it is safe to use the fileno.

**Comment by rickardp** [comment]
Fixed in PR #10 now that I finally migrated to GitHub actions
